client/src/pages/Home/DebugBar.tsx

```diff
@@ -44,90 +47,53 @@ function DebugBar({
   const grabLoc = () => {
-    // Use browser navigator if we are in web environment & available
-    if (isWeb) {
-      if (navigator.geolocation) {
-        setIsLoading(true);
-        navigator.geolocation.getCurrentPosition(fetchStores, errorbackLoc);
-      }
-    } else {
-      setIsLoading(true);
-      microapps
-        .getCurrentLocation()
-        .then((loc: any) => {
-          const position = {
-            coords: {
-              latitude: loc["latitude"],
-              longitude: loc["longitude"],
-            },
-          };
-          fetchStores(position);
-        })
-        .catch((err: any) => errorbackLoc(err));
-    }
+    setIsLoading(true);
+    getGeoLocation(geoLocationCallback);
   };
 
-  // fetch list of users
-  const fetchStores = async (position: {
-    coords: {
-      latitude: number;
-      longitude: number;
-    };
-  }) => {
-    // Update location
-    const data = {
-      distance: 10000,
-      latitude: position.coords.latitude,
-      longitude: position.coords.longitude,
-    };
-    const stores = await fetchJson("POST", data, STORE_LIST_API);
-    setUserCoords([position.coords.latitude, position.coords.longitude]);
-    storesCallback(stores);
+  const geoLocationCallback = (position: GeoLocation) => {
     setIsLoading(false);
+    setCurGeoLocation(position);
+  };
+
+  const fetchStores = async (position?: GeoLocation) => {
+    // When called without arguments, we default to using
+    // already present location previously retrieved
+    if (!position) {
+      if (curGeoLocation) {
+        position = curGeoLocation;
+      } else {
+        return;
+      }
+    }
+    getStoresByLocation(position).then((res: any) => storesCallback(res));
   };
 
   // Request PlacesAPI for nearby locations
   const getNearbyPlaces = () => {
-    const curLoc = new google.maps.LatLng(userCoords[0], userCoords[1]);
-    const request = {
-      location: curLoc,
-      radius: PLACES_RADIUS_METERS,
-      type: PLACES_TYPES,
-    };
-    setIsLoading(true);
-    service.nearbySearch(request, getNearbyPlacesCallback);
+    if (curGeoLocation) {
+      setIsLoading(true);
+      getNearbyPlacesTest(curGeoLocation, getNearbyPlacesCallback);
+    }
   };
 
   // Digest PlacesAPI results
   const getNearbyPlacesCallback = (places: any, status: any) => {
+    setIsLoading(false);
     if (status === google.maps.places.PlacesServiceStatus.OK) {
-      setNearbyPlaces(places);
-    } else {
-      alert("PlacesAPI failure to return");
+      if (placesCallback) {
+        placesCallback(places);
+      }
     }
-    setIsLoading(false);
   };
 
   // Get user identity details
-  const loginUser = async () => {
-    // Only run in iframe (on app)
-    if (!isWeb) {
-      const request = { nonce: "Don't Hack me please" };
-      microapps.getIdentity(request).then((response: any) => {
-        setIdentity(extractIdentityToken(response));
-      });
+  const testLogin = async () => {
```

Comment:
why is this called testLogin?

Reply:
The DebugBar component is only ever used for testing functionalities, at this point in the normal flow (/home), our user is already logged in.